origins_workflow/src/Controller/AuditController.php

```diff
@@ -0,0 +1,139 @@
+   *   The current user.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, LoggerInterface $logger, AccountInterface $account) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->logger = $logger;
+    $this->account = $account;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('logger.factory')->get('origins_workflow'),
+      $container->get('current_user')
+    );
+  }
+
+  /**
+   * Content Audit.
+   *
+   * @return string
+   *   Return confirmation string.
+   */
+  public function contentAudit($nid) {
+    $render_array = [];
+    $msg = $this->t('Invalid node ID - @nid', ['@nid' => $nid]);
+    if (!empty($nid)) {
+      $node = $this->entityTypeManager()->getStorage('node')->load($nid);
+      if ($node) {
+        // Retrieve audit text from config.
+        $audit_button_text = $this->config('origins_workflow.auditsettings')->get('audit_button_text');
+        $audit_confirmation_text = $this->config('origins_workflow.auditsettings')->get('audit_confirmation_text');
+        // Show confirmation text to user.
+        $render_array['confirmation_text'] = [
+          '#markup' => $this->t($audit_confirmation_text),
+          '#prefix' => "<div class='confirmation_text'>",
+          '#suffix' => "</div>",
+          '#weight' => 0,
+        ];
+        // Build a confirm link.
+        $link_object = Link::createFromRoute($this->t($audit_button_text),
+          'origins_workflow.audit_controller_confirm_audit',
+          ['nid' => $nid],
+          ['attributes' => ['rel' => 'nofollow', 'class' => 'audit_link']]);
+        // Add confirm link to render array.
+        $render_array['link1'] = array_merge($link_object->toRenderable(),
+          [
+            '#prefix' => "<span class='confirm_audit'>",
+            '#suffix' => "</span>",
+            '#weight' => 1,
+          ]);
+        // Build a cancel link.
+        $link_object_cancel = Link::createFromRoute($this->t("Cancel"),
+          'entity.node.canonical',
+          ['node' => $nid],
+          ['attributes' => ['rel' => 'nofollow', 'class' => 'cancel_link']]);
+        // Add cancel link to render array.
+        $render_array['link2'] = array_merge($link_object_cancel->toRenderable(),
+          [
+            '#prefix' => "<span class='cancel'>",
+            '#suffix' => "</span>",
+            '#weight' => 2,
+          ]);
+      }
+    }
+    return $render_array;
+  }
+
+  /**
+   * Confirm Audit.
+   *
+   * @return string
+   *   Return confirmation string.
+   */
+  public function confirmAudit($nid) {
+    // Bump up the 'next audit due' date and log it.
+    $node = $this->entityTypeManager()->getStorage('node')->load($nid);
+    if ($node) {
+      $node->set('field_next_audit_due', date('Y-m-d', strtotime("+6 months")));
```

Comment:
Will this always be 6 months?

Reply:
Good idea, but have added TODO and will implement later